extensions/resteasy-reactive/rest-client/deployment/src/test/java/io/quarkus/rest/client/reactive/ContextLocalPropagationTest.java

```diff
@@ -0,0 +1,150 @@
+            this.client = client;
+        }
+
+        @Path("invokeClient")
+        @GET
+        public String invokeClient() {
+            var result = client.get();
+            Optional<String> fromRequest = ContextLocals.get("fromRequest");
+            Optional<String> fromResponse = ContextLocals.get("fromResponse");
+            return result + "/" + fromRequest.orElse("none") + "/" + fromResponse.orElse("none");
+        }
+
+        @Path("client-inherits-context")
+        @GET
+        public Uni<String> clientInheritsContext() {
+            ContextLocals.put("parent-value", "set-in-parent");
+            return client.getUni()
+                    .map(result -> result + "/" + ContextLocals.get("parent-value").orElse("none"));
+        }
+
+        @Path("toClient")
+        @GET
+        public String toClient() {
+            return "test";
+        }
+
+        @Path("toClient2")
+        @GET
+        public String toClient2() {
+            return "test";
+        }
+    }
+
+    @Path("test")
+    @RegisterRestClient(configKey = "client")
+    @RegisterProvider(RequestFilter.class)
+    @RegisterProvider(ResponseFilter.class)
+    public interface Client {
+
+        @GET
+        @Path("toClient")
+        String get();
+
+        @GET
+        @Path("toClient2")
+        Uni<String> getUni();
+    }
+
+    public static class RequestFilter implements ClientRequestFilter {
+
+        @Override
+        public void filter(ClientRequestContext requestContext) throws IOException {
+            RestClientContextUtil.putLocal("fromRequest", "foo");
+        }
+    }
+
+    public static class ResponseFilter implements ClientResponseFilter {
+
+        @Override
+        public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext)
+                throws IOException {
+            RestClientContextUtil.putLocal("fromResponse", "bar");
+        }
+    }
+
+    public static final class RestClientContextUtil {
+
+        private RestClientContextUtil() {
+        }
+
+        public static void putLocal(Object key, Object value) {
+            determineRestClientContext().putLocal(key, value);
+        }
+
+        private static Context determineRestClientContext() {
+            // In an ideal world, this would always be populated, however because we never
+            // defined a proper execution model for the REST Client handlers, currently we are
+            // in a situation where request filters could be run on the calling context
+            // and not the client's purpose built context.
+            // We will need a proper solution soon, but as we need to have a proper way to
+            // set contextual information in Quarkus 3.20 (LTS), we can't risk breaking
+            // client code everywhere, so for now we will tell people to check the context
+            Optional<Object> maybeParentContext = ContextLocals.get("__PARENT_CONTEXT__");
```

Comment:
This is the only change I made in the test as the `getParentContext()` API is not there.